test/integration/transactions.js

```diff
@@ -258,6 +259,27 @@ describe('bitcoinjs-lib (transactions)', function () {
     })
   })
 
+  // CAUTION: if you don't understand what this example is for, don't do it
+  it('can sign a Transaction with something other than ECPair', function () {
+    const txb = new bitcoin.TransactionBuilder()
+
+    txb.setVersion(1)
+    txb.addInput('61d520ccb74288c96bc1a2b20ea1c0d5a704776dd0164a396efec3ea7040349d', 0) // Alice's previous transaction output, has 15000 satoshis
+    txb.addOutput('1cMh228HTCiwS8ZsaakH8A8wze1JR5ZsP', 12000)
+    // (in)15000 - (out)12000 = (fee)3000, this is the miner fee
+
+    txb.sign(0, {
+      publicKey: Buffer.from('029f50f51d63b345039a290c94bffd3180c99ed659ff6ea6b1242bca47eb93b59f', 'hex'),
+      sign: function (hash) {
+        const privateKey = Buffer.from('8c112cf628362ecf4d482f68af2dbb50c8a2cb90d226215de925417aa9336a48', 'hex')
+        return tinysecp.sign(hash, privateKey)
```

Comment:
You should add a comment here saying what format this sign function expects.

```javascript
// sign expects a 64 byte Buffer where the first and last 32 bytes correspond with the r and s values respectively of the ECDSA signature
```

etc.

Reply:
expects meaning "expects as a return value"

sign referring to line 273, the function we are declaring here.

Reply:
@junderw can you add it? :+1: